### PR TITLE
Fix: Switching between multi-extruder printers using tabs can cause t…

### PR DESCRIPTION
**Pull request by womendoushihaoyin**
# Reproduction steps for the issue:
1. Select multiple multi-tool printers at once. （eg: prusa x5l /  Snapmaker j1/Artisan）
image
2. Switch to another multi-tool printer using the tab.
image

you can observe that only one tool remains. Moreover, for multi-tool machines, Orca no longer allows the addition or removal of filament buttons (default). This can be very confusing for users.

# Analysis：
1. In Orca, there is an inconsistency in how the extruder_count is determined for multi-tool printers. Some parts of the code use nozzle_diameter to determine the number of extruders, while others use extruder_colour 
. This inconsistency can lead to confusion and errors in the user interface, such as the addition or removal of filament buttons not being available for multi-tool machines, which will be perplexing.

To address this issue and align with the characteristics of printers like Prusa and Snapmaker, it is recommended to standardize the determination of extruder_count by using nozzle_diameter for parsing across the software. This approach would ensure a unified method for identifying the number of extruders, thereby reducing confusion and improving the user experience.

2. In multi-tool scenarios, modify the parameters passed to on_filaments_change.

3. prevent crashing

### CODE REVIEW

**Review by SoftFever — COMMENTED** [review]
Very nice catch!
Thank you!

File: src/slic3r/GUI/Tab.cpp [hunk 1]
```diff
@@ -6079,8 +6085,9 @@ void Page::update_visibility(ConfigOptionMode mode, bool update_contolls_visibil
 #ifdef __WXMSW__
     if (!m_show) return;
     // BBS: fix field control position
-    wxTheApp->CallAfter([this]() {
-        for (auto group : m_optgroups) {
+    auto groups = this->m_optgroups;
```
Comment (SoftFever) [inline src/slic3r/GUI/Tab.cpp:6088]: @womendoushihaoyin 
Is this change intended? Any reason to change it?

Comment (womendoushihaoyin) [reply src/slic3r/GUI/Tab.cpp:6088]: Yes, this change is necessary. If "this" is captured, it will cause unexplained occasional crashes.

**Review by Noisyfox — APPROVED** [review]
LGTM

**Comment by womendoushihaoyin** [comment]
Please temporarily ignore this PR, as it may cause some other issues. I will provide a more reasonable solution shortly.

**Comment by Noisyfox** [comment]
So should I revert this until then?

**Comment by womendoushihaoyin** [comment]
I'm afraid so, I will submit a new pull request as soon as possible.

**Comment by womendoushihaoyin** [comment]
https://github.com/SoftFever/OrcaSlicer/pull/7860
